CMakeLists.txt

```diff
@@ -1,7 +1,44 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(kimera_vio_ros)
 
-find_package(catkin_simple REQUIRED)
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+find_package(ament_cmake REQUIRED)
+find_package(rclcpp REQUIRED)
+
+find_package(geometry_msgs REQUIRED)
+find_package(nav_msgs REQUIRED)
+find_package(pcl_msgs)
+find_package(pose_graph_msgs)
+find_package(sensor_msgs REQUIRED)
+find_package(std_msgs REQUIRED)
+
+find_package(cv_bridge REQUIRED)
+find_package(tf2_ros REQUIRED)
+
+# find_package(Gflags REQUIRED)
+# find_package(Glog 0.3.5 REQUIRED)
+find_package(GTSAM REQUIRED)
+# find_package(opengv REQUIRED)
+find_package(OpenCV 3.3.1 REQUIRED)
+# find_package(DBoW2 REQUIRED)
+find_package(kimera_vio REQUIRED)
+find_package(PCL REQUIRED QUIET COMPONENTS common io)
+
+find_package(pose_graph_visualizer)
+
+#find_package(pcl_ros)
+find_package(pcl_conversions)
+
+find_package(message_filters)
+find_package(image_transport)
```

Comment:
@ruffsl I am not familiar with ROS2, but is this necessary? Or even, is this standard?
Why not using catkin_simple?

Reply:
The officially supported method for integrating ROS2 with cmake is ament, please see:
https://design.ros2.org/articles/ament.html

It's hygienic to explicitly declare which dependencies are to be used in cmake, rather than roping just anything in the package.xml. This can be an issue when the package dependency name, and library the depency exports are not perfectly aligned strings. Also, it seems that catkin_simple hasn't been maintained since 2015.

Reply:
I see, `ament_cmake` seems like a big deal. Thanks for the pointer.